### PR TITLE
Run on stable, enable build cache in CI

### PR DESCRIPTION
**Pull request by Alexendoo**
Saves a good minute+ on each action

This also means locally you don't have to rebuild anything to run glacier again the next day

### CODE REVIEW

File: .github/actions/build/action.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+runs:
+  using: "composite"
+  steps:
+    - name: Configure rustup
+      id: rustup
+      run: |
+        rustup update
+        rustup toolchain install --profile minimal nightly
+        rustup target add --toolchain nightly x86_64-pc-windows-msvc
+      shell: bash
```
Comment (JohnTitor) [inline .github/actions/build/action.yml:12]: IIRC the default value of `shell` on the ubuntu runner is `bash` so we don't need this.

File: .github/actions/build/action.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        restore-keys: build-
+
+    # We restore source files from the cache to keep their modification times,
+    # then use git to undo that for files that have actually changed. This
+    # prevents unnecessary rebuilds
+    - run: |
+        git status
+        git checkout .
+        git clean -f .
+      shell: bash
```
Comment (JohnTitor) [inline .github/actions/build/action.yml:33]: Btw syntax seems wrong.

**Review by JohnTitor — APPROVED** [review]
Great idea! I'd like to write down "you need a nightly toolchain to run" in README but otherwise LGTM.

File: .github/actions/build/action.yml (hunk 1 above)
Comment (Alexendoo) [reply .github/actions/build/action.yml:12]: It does feel redundant but apparently in actions you must specify the shell, wouldn't run without it

**Comment by Alexendoo** [comment]
Good point on the README, I will add that

File: .github/actions/build/action.yml (hunk 2 above)
Comment (Alexendoo) [reply .github/actions/build/action.yml:33]: How so? I can't see it

File: .github/actions/build/action.yml (hunk 1 above)
Comment (JohnTitor) [reply .github/actions/build/action.yml:12]: Ah, so a template-like action needs it? Then totally fine.

File: .github/actions/build/action.yml (hunk 2 above)
Comment (JohnTitor) [reply .github/actions/build/action.yml:33]: Oh well GitHub's highlighting seems wrong then:
image
Just checked on my editor and there's no problem, weird...

Comment (Alexendoo) [reply .github/actions/build/action.yml:33]: ooh I see it now! tricky to spot with the light theme

image

Yeah, odd, guess it's a GH bug

**Comment by JohnTitor** [comment]
Looks great, feel free to merge if you're ready :)

**Pull request merged**